posts/2024-09-05-impl-trait-capture-rules.md

```diff
@@ -0,0 +1,211 @@
+---
+layout: post
+title: "Changes to impl Trait in Rust 2024"
+author: Niko Matsakis
+team: the language team <https://www.rust-lang.org/governance/teams/lang>
+---
+This blog post describes some small but significant changes with (return position) `impl Trait` capture rules that are coming in Rust 2024. The goal of these changes is to simplify how `impl Trait` works to better match what people want while also giving a flexible syntax that allows users to have full control when needed.
+
+## TL;DR
+
+* Return-position impl Trait in Rust 2024 can now reference all fn arguments by default.
+* We are introducing a new syntax (`+ use<>`) that you can use to specify precisely which generic parameters can appear in the hidden type.
+
+## Background: return position impl trait
+
+This blog post concerns *return position `impl Trait`*, such as the following example:
+
+```rust
+fn process_datums(
+    datums: &[Datum]
+) -> impl Iterator<Item = ProcessedDatum> {
+    datums
+        .iter()
+        .map(|datum| datum.process())
+}
+```
```

Comment:
The plural of "datum" is "data".

Reply:
I'm aware of that :) I just like `s`. But I guess I'll change it...